src/main/java/com/google/sps/servlets/DestinationDataServlet.java

```diff
@@ -0,0 +1,147 @@
+            .withLng(Double.parseDouble(request.getParameter("longitude")))
+            .build();
+
+    String city = request.getParameter("city");
+    String description = request.getParameter("description");
+
+    Riddle riddle =
+        new Riddle.Builder()
+            .withPuzzle(request.getParameter("riddle"))
+            .withHint(request.getParameter("hint1"))
+            .withHint(request.getParameter("hint2"))
+            .withHint(request.getParameter("hint3"))
+            .build();
+
+    /* Retrieves the obscurity level chosen by the user as a List of Strings and converts the level to an Enum Obscurity value */
+    List<String> levels =
+        Arrays.stream(request.getParameterValues("obscurity"))
+            .filter(level -> level != null)
+            .collect(Collectors.toList());
+    Destination.Obscurity level = convertLevelToEnum(levels);
+
+    /* Retrieves the tags selected by the user as a List of Strings and converts them into a Set of Enum Tags */
+    List<String> tags =
+        Arrays.stream(request.getParameterValues("tag"))
+            .filter(tag -> tag != null)
+            .collect(Collectors.toList());
+    Set<Destination.Tag> checkedTags = convertTagsToEnum(tags);
+
+    Destination d1 =
+        new Destination.Builder()
+            .withName(name)
+            .withLocation(location)
+            .withCity(city)
+            .withDescription(description)
+            .withRiddle(riddle)
+            .withTags(checkedTags)
+            .withObscurity(level)
+            .build();
+
+    destinations.add(d1);
+    response.sendRedirect("/destination-data");
+  }
+
+  /* Retrieves the destination created by the user, turns it into a JSON formatted string,
+   *and displays the JSON-ified String on /destination-data
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    String json = gson.toJson(destinations.get(destinations.size() - 1));
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  public Set<Destination.Tag> convertTagsToEnum(List<String> tags) {
+    Set<Destination.Tag> tagEnums = new HashSet<Destination.Tag>();
+    for (String tag : tags) {
+      switch (tag) {
+        case "art":
+          tagEnums.add(Destination.Tag.ART);
+          break;
+        case "sports":
+          tagEnums.add(Destination.Tag.SPORT);
+          break;
+        case "historical":
+          tagEnums.add(Destination.Tag.HISTORICAL);
+          break;
+        case "food":
+          tagEnums.add(Destination.Tag.FOOD);
+          break;
+        case "family":
+          tagEnums.add(Destination.Tag.FAMILY);
+          break;
+        case "tourist":
+          tagEnums.add(Destination.Tag.TOURIST);
+          break;
+        default:
+          tagEnums.add(Destination.Tag.UNDEFINED);
```

Comment:
i don't think you want to add UNDEFINED to your set of tags, right?

Reply:
That is correct. From what I remember from our conversation last Friday, you had said that this was a good practice for Enums. I can remove this if that is not the case.

Reply:
Yes, it's good practice to have an UNDEFINED value when you define the enum.

In this particular case, you're taking in a list of strings the user has given you and converting it to a set of enums.  Since the user isn't ever going to be specifying "undefined" as a tag, there's no point in adding it to your enum set.

Reply:
Oh ok that makes sense. I will remove it